core-web/libs/block-editor/src/lib/plugins/bubble-link-form.plugin.ts

```diff
@@ -230,35 +239,94 @@ export class BubbleLinkFormView {
+    private hanlderScroll(e: Event) {
+        const element = e.target as HTMLElement;
+        const parentElement = element?.parentElement?.parentElement;
+        // If text is too long, the input fires the `scroll` event.
+        // When that happens, we do not want to hide the tippy.
+        if (this.scrollElementMap[element.id] || this.scrollElementMap[parentElement.id]) {
+            return;
         }
+        this.hide();
     }
 }
 
 export const bubbleLinkFormPlugin = (options: BubbleLinkFormProps) => {
+    let lastNode;
     return new Plugin({
         key: options.pluginKey as PluginKey,
         view: (view) => new BubbleLinkFormView({ view, ...options }),
         state: {
             init(): PluginState {
                 return {
-                    toggle: true
+                    isOpen: false,
+                    openOnClick: false
                 };
             },
 
-            apply(transaction: Transaction): PluginState {
-                const transactionMeta = transaction.getMeta(options.pluginKey);
-                if (transactionMeta) {
+            apply(
+                transaction: Transaction,
+                value: PluginState,
+                oldState: EditorState
+            ): PluginState {
+                const { isOpen, openOnClick } = transaction.getMeta(LINK_FORM_PLUGIN_KEY) || {};
+                const state = LINK_FORM_PLUGIN_KEY.getState(oldState);
+
+                if (typeof isOpen === 'boolean') {
                     return {
-                        toggle: options.storage.show
+                        isOpen,
+                        openOnClick
                     };
                 }
 
-                return {
-                    toggle: options.storage.show
-                };
+                // keep the old state in case we do not receive a new one.
+                return state || value;
+            }
+        },
+        props: {
+            handleDOMEvents: {
+                mousedown(view, event) {
+                    const editor = options.editor;
+                    const pos = getPosAtDocCoords(view, event);
+                    const { isOpen, openOnClick } = LINK_FORM_PLUGIN_KEY.getState(editor.state);
+
+                    // Prevent to open the bubble Menu
+                    // After closing the link form on click.
+                    if (isOpen && openOnClick) {
+                        editor.chain().unsetHighlight().setTextSelection(pos).run();
+                    }
+                }
+            },
+            handleClickOn(view: EditorView, pos: number, node) {
+                const editor = options.editor;
+
+                if (!editor.isActive('link') || !pos) {
+                    lastNode = node;
+                    return;
+                }
+
+                // If we click again in the same link node,
+                // We close the form and enable editing.
+                if (JSON.stringify(lastNode) === JSON.stringify(node)) {
```

Comment:
Is this safe enough? can't the order of the object mess this?

Reply:
Great catch!